### PR TITLE
api: MinioClient now takes secure parameter instead of insecure.

### LINKED ISSUE #375: Change insecure to secure variable
According to our discussions : If language supports default params without asking the user to pass them, use it. Otherwise force user to pass the default.

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #375

### CODE REVIEW

File: src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -301,16 +301,16 @@ public MinioClient(HttpUrl url, String accessKey, String secretKey)
...
    */
   public MinioClient(String endpoint, int port, String accessKey, String secretKey)
     throws InvalidEndpointException, InvalidPortException {
-    this(endpoint, port, accessKey, secretKey, false);
+    this(endpoint, port, accessKey, secretKey, true);
   }
 
   /**
-   * Creates Minio client object with given endpoint, access key and secret key using insecure (HTTP) connection.
+   * Creates Minio client object with given endpoint, access key and secret key using secure (HTTP) connection.
```
Comment (balamurugana) [inline src/main/java/io/minio/MinioClient.java]: Please change HTTP to HTTPS

File: src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -330,23 +330,23 @@ public MinioClient(String endpoint, int port, String accessKey, String secretKey
    *
    * @param accessKey Access key to access service in endpoint.
    * @param secretKey Secret key to access service in endpoint.
-   * @param insecure  If true, access endpoint using HTTP else access it using HTTPS.
+   * @param secure If false, access endpoint using HTTP else access it using HTTPS.
```
Comment (balamurugana) [inline src/main/java/io/minio/MinioClient.java]: Please reword to have positive case first like
"If true, access endpoint using HTTPS else access it using HTTP"

File: src/main/java/io/minio/MinioClient.java [hunk 3]
```diff
@@ -367,22 +367,22 @@ public MinioClient(String endpoint, String accessKey, String secretKey, boolean
    * @param port      Valid port.  It should be in between 1 and 65535.  Unused if endpoint is an URL.
    * @param accessKey Access key to access service in endpoint.
    * @param secretKey Secret key to access service in endpoint.
-   * @param insecure  If true, access endpoint using HTTP else access it using HTTPS.
+   * @param secure  If false, access endpoint using HTTP else access it using HTTPS.
```
Comment (balamurugana) [inline src/main/java/io/minio/MinioClient.java]: same as above

**Comment by balamurugana** [comment]
Very minor comments

**Comment by harshavardhana** [comment]
> Very minor comments

Done!

**Comment by balamurugana** [comment]
:+1: LGTM

**Pull request merged**